syntax = "proto3";

package fp_network;

message HostState {
    enum State {
        DISCONNECTING = 0;
    }
    State state = 1;
}

message VideoFrame {
    enum FrameType {
        NORMAL = 0;
        IDR = 1;
        PPS_SPS = 2;
    }
    FrameType frame_type = 1;
    uint32 chunk_offset = 2;
    oneof DataBacking {
        bytes data = 3;
        // For actor->actor messages
        uint64 data_handle = 4;
    }
}

message AudioFrame {
    uint32 chunk_offset = 1;
    oneof DataBacking {
        bytes data = 2;
        // For actor->actor messages
        uint64 data_handle = 3;
    }
}

message HostDataFrame {
    uint32 frame_num = 1;
    uint32 frame_size = 2;
    uint32 stream_point = 3;
    oneof DataFrame {
        VideoFrame video = 4;
        AudioFrame audio = 5;
    }
}